web/modules/custom/joinup_subscription/src/Form/UnsubscribeFromAllForm.php

```diff
@@ -55,90 +56,82 @@ class UnsubscribeFromAllForm extends ConfirmFormBase {
    *
    * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
    *   The entity type manager service.
-   * @param \Drupal\og\MembershipManagerInterface $membership_manager
-   *   The membership manager service.
    * @param \Drupal\Core\Render\Renderer $renderer
    *   The renderer service.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $bundle_info
+   *   The entity type bundle info service.
    */
-  public function __construct(EntityTypeManagerInterface $entity_type_manager, MembershipManagerInterface $membership_manager, Renderer $renderer) {
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, Renderer $renderer, EntityTypeBundleInfoInterface $bundle_info) {
     $this->entityTypeManager = $entity_type_manager;
-    $this->membershipManager = $membership_manager;
     $this->renderer = $renderer;
+    $this->bundleInfo = $bundle_info;
   }
 
   /**
    * {@inheritdoc}
    */
-  public static function create(ContainerInterface $container) {
+  public static function create(ContainerInterface $container): self {
     return new static(
       $container->get('entity_type.manager'),
-      $container->get('og.membership_manager'),
-      $container->get('renderer')
+      $container->get('renderer'),
+      $container->get('entity_type.bundle.info')
     );
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getFormId() {
+  public function getFormId(): string {
     return 'unsubscribe_from_all_form';
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getQuestion() {
-    return $this->t('Unsubscribe from all :bundles', [
-      ':bundle' => $this->bundle,
-    ]);
+  public function getQuestion(): MarkupInterface {
+    return $this->t('Unsubscribe from all?');
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getDescription() {
-    return $this->bundle === 'collection'
-      ? $this->t('Are you sure you want to unsubscribe from all collections?<br />You will stop receiving news and updates from all collections (including the pending memberships).<br />In the future you will not be notified for the following collections:')
-      : $this->t('Are you sure you want to unsubscribe from all solutions?<br />You will stop receiving news and updates from all solutions.<br />In the future you will not be notified for the following solutions:');
+  public function getDescription(): MarkupInterface {
+    return $this->t('Are you sure you want to unsubscribe from all collections and/or solutions? You will stop receiving news and updates, including the pending memberships, from the following:');
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getCancelUrl() {
-    return Url::fromRoute('joinup_subscription.my_subscriptions', [
-      'subscription_type' => $this->bundle,
-    ]);
+  public function getCancelUrl(): Url {
+    return Url::fromRoute('joinup_subscription.my_subscriptions');
```

Comment:
The only remark I found is this cancel URL. When a moderator performs the unsubscribe-from-all action in the profile of another user, the cancel URL as well as unsubscribing the user from all groups, redirects the user to their own subscription page instead of the user they were performing the action at.